feat(core): Better handling of versioned FHIR resource properties

apeteri opened the pull request:
https://snowowl.atlassian.net/browse/SNRAY-1095

apeteri committed:
feat(core): Add snapshot of resource properties to VersionDocument

apeteri committed:
Merge remote-tracking branch 'origin/8.x' into feature/fhir-versioned-resource-properties

apeteri committed:
refactor(core): Snapshotting does not need to work on the document level

apeteri committed:
feat(core): Retrieve resource document with matching "created" timestamp

Time-appropriate information is only copied over to the ResourceFragment
instance if it did not already contain it.

apeteri committed:
fix(core): Resolve remaining issues with FhirResourceSearchRequest

cmark commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/ResourceDocument.java:
```diff
@@ -215,6 +212,13 @@ public static Expression upgradeOfs(Iterable<ResourceURI> upgradeOfs) {
 			return matchAny(Fields.UPGRADE_OF, Collections3.toImmutableSet(upgradeOfs).stream().map(ResourceURI::toString).collect(Collectors.toSet()));
 		}
 		
+		public static Expression validAsOf(Long createdTimestamp) {
```
Can we move this to the `Revision` class?

cmark reviewed (changes requested):
Early review.

apeteri committed:
fix(core): Index resource snapshot fields on VersionDocument

This generates docValues for each field, allowing them to participate in
quick field selection.

apeteri committed:
refactor(core): Move simple revision range filter expression to Revision

cmark reviewed (approved):
🥓

nagyo reviewed (approved):
semi-RSLGTM!
